File: src/go/partition/lib/partitions2json.go
package partition

import (
	"encoding/json"
	"fmt"
	"os"
)

func WritePartitions(result []int32, resultFile string) {
	file, _ := os.Create(resultFile)

	// Convert to bool array
	var resultBool []bool
	for _, v := range result {
		if v == 1 {
			resultBool = append(resultBool, true)
		} else {
			resultBool = append(resultBool, false)
		}
	}

	encoder := json.NewEncoder(file)
	if err := encoder.Encode(resultBool); err != nil {
		fmt.Println("Error encoding JSON:", err)
		return
	}
}

File: src/julia/partition/partitions2json.jl
module Partitions2Json

using JSON

export partitions2json
function partitions2json(file, arr::Vector{Bool})
    jsonString = JSON.json(arr)
    write(file, jsonString)
end # partitions2json

export writePartitions
function writePartitions(file, arr::Vector{Bool})
    partitions2json(file, arr)
end

end # module

